orchagent/routeorch.cpp

```diff
@@ -800,25 +824,29 @@ void RouteOrch::doTask(Consumer& consumer)
                     }
                     else if (srv6_nh == true)
                     {
-                        string ip;
-                        if (ipv.empty())
+                        if (srv6_vpn && (srv6_vpn_sidv.size() != srv6_src.size()))
                         {
-                            ip = "0.0.0.0";
+                            SWSS_LOG_ERROR("inconsistent number of endpoints and srv6 vpn sids.");
+                            it = consumer.m_toSync.erase(it);
+                            continue;
                         }
-                        else
+
+                        if (srv6_seg && (srv6_segv.size() != srv6_src.size()))
                         {
-                            SWSS_LOG_ERROR("For SRV6 nexthop ipv should be empty");
+                            SWSS_LOG_ERROR("inconsistent number of srv6_segv and srv6_srcs.");
                             it = consumer.m_toSync.erase(it);
                             continue;
                         }
-                        nhg_str = ip + NH_DELIMITER + srv6_segv[0] + NH_DELIMITER + srv6_src[0];
 
-                        for (uint32_t i = 1; i < srv6_segv.size(); i++)
+                        for (uint32_t i = 0; i < srv6_src.size(); i++)
                         {
-                            nhg_str += NHG_DELIMITER + ip;
-                            nhg_str += NH_DELIMITER + srv6_segv[i];
-                            nhg_str += NH_DELIMITER + srv6_src[i];
+                            if (i) nhg_str += NHG_DELIMITER;
+                            nhg_str += (ipv.size() > i ? ipv[i] : "0.0.0.0") + NH_DELIMITER;  // ip address
+                            nhg_str += (srv6_seg ? srv6_segv[i] : "") + NH_DELIMITER;     // srv6 segment
+                            nhg_str += srv6_src[i] + NH_DELIMITER; // srv6 source
+                            nhg_str += (srv6_vpn ? srv6_vpn_sidv[i] : "") + NH_DELIMITER; // srv6 vpn sid
```

Comment:
similar comment to NHKey, do we need NH_DELIMITER after VPN SID?

Reply:
As mentioned above, we'd better have.